lib/ae_mdw/db/sync/transaction.ex

```diff
@@ -128,14 +128,40 @@ defmodule AeMdw.Db.Sync.Transaction do
       Sync.Contract.events(events, txi, ct_txi)
     end
 
-    for {field, pos} <- AE.tx_ids(type) do
-      <<_::256>> = pk = resolve_pubkey(elem(tx, pos), type, field, block_index)
-      write_field(type, pos, pk, txi)
+    case type do
+      :spend_tx ->
+        sync_spend_tx(tx, txi, block_index)
+
+      _ ->
+        for {field, pos} <- AE.tx_ids(type) do
+          <<_::256>> = pk = resolve_pubkey(elem(tx, pos), type, field, block_index)
+          write_field(type, pos, pk, txi)
+        end
     end
 
     txi + 1
   end
 
+  defp sync_spend_tx(tx, txi, block_index) do
+    ids = AE.tx_ids(:spend_tx)
+
+    <<_::256>> =
+      sender_pk = resolve_pubkey(elem(tx, ids.sender_id), :spend_tx, :sender_id, block_index)
+
+    <<_::256>> =
+      recipient_pk =
+      resolve_pubkey(elem(tx, ids.recipient_id), :spend_tx, :recipient_id, block_index)
+
+    case sender_pk == recipient_pk do
+      true ->
+        write_field(:spend_tx, ids.sender_id, sender_pk, txi)
+
+      false ->
+        write_field(:spend_tx, ids.sender_id, sender_pk, txi)
+        write_field(:spend_tx, ids.recipient_id, recipient_pk, txi)
+    end
```

Comment:
sync_spend_tx could return :ok to be explicit we don't need the result

Reply:
it would help with the dialyzer complain

Reply:
actually did not because Function sync_spend_tx/3 has no local return. is due to tx_ids as you already noticed.

Reply:
but the :ok return would be welcome (: